File: app/src/main/java/example/tablayout/TabLayoutActivity.java
package example.tablayout;

import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;

import com.style.base.BaseActivity;
import com.style.framework.R;
import com.style.framework.databinding.ActivityTabLayoutBinding;

import java.util.ArrayList;
import java.util.List;

public class TabLayoutActivity extends BaseActivity {

    ActivityTabLayoutBinding bd;
    private FindTabAdapter fAdapter;                               //定义adapter

    private List<Fragment> fragments = new ArrayList<>();                                //定义要装fragment的列表
    private List<String> titles = new ArrayList<>();
    @Override
    public int getLayoutResId() {
        return R.layout.activity_tab_layout;
    }

    @Override
    public void initData() {
        bd = getBinding();
        setToolbarTitle("tabLayout");
        for (int i = 1; i < 10; i++) {
            titles.add("title_" + i);
            fragments.add(TabLayoutFragment.newInstance(i + ""));
        }
        fAdapter = new FindTabAdapter(this.getSupportFragmentManager(), fragments, titles);
        bd.viewPager.setAdapter(fAdapter);
        bd.tabLayout.setupWithViewPager(bd.viewPager);
        bd.tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {

            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {

            }

            @Override
            public void onTabReselected(TabLayout.Tab tab) {

            }
        });
    }
}

File: app/src/main/java/example/tablayout/TabLayoutActivity.kt
package example.tablayout

import android.support.design.widget.TabLayout
import android.support.v4.app.Fragment

import com.style.base.BaseTitleBarActivity
import com.style.framework.R
import com.style.framework.databinding.ActivityTabLayoutBinding

import java.util.ArrayList

class TabLayoutActivity : BaseTitleBarActivity() {

    private lateinit var bd: ActivityTabLayoutBinding
    private lateinit var fAdapter: FindTabAdapter                            //定义adapter
    private val fragments = ArrayList<Fragment>()                                //定义要装fragment的列表
    private val titles = ArrayList<String>()

    override fun getLayoutResId(): Int {
        return R.layout.activity_tab_layout
    }

    override fun initData() {
        bd = getBinding()
        setToolbarTitle("tabLayout")
        titles.add("系统挂件")
        fragments.add(SystemWidgetFragment())
        titles.add("title_0")
        fragments.add(TabLayoutFragment.newInstance("BottomSheetBehavior"))
        titles.add("title_1")
        fragments.add(BottomSheetDialogFragment.newInstance("BottomSheetDialog"))
        fAdapter = FindTabAdapter(this.supportFragmentManager, fragments, titles)
        bd.viewPager.adapter = fAdapter
        bd.tabLayout.setupWithViewPager(bd.viewPager)
        bd.tabLayout.addOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
            override fun onTabSelected(tab: TabLayout.Tab) {

            }

            override fun onTabUnselected(tab: TabLayout.Tab) {

            }

            override fun onTabReselected(tab: TabLayout.Tab) {

            }
        })
    }
}
